torchmetrics/metric.py

```diff
@@ -221,6 +222,46 @@ def wrapped_func(*args, **kwargs):
 
         return wrapped_func
 
+    @contextmanager
+    def _apply_sync(
+        self,
+        dist_sync_fn: Optional[Callable] = None,
+    ) -> None:
+        """
+        Context manager to synchronize the states between processes when running in a distributed setting
+        and restore the local cache states after yielding.
+
+        Args:
+            dist_sync_fn: Function to be used to perform states synchronization
+        """
+
+        is_distributed = torch.distributed.is_available() and torch.distributed.is_initialized()
+
+        if not is_distributed:
+            yield
+            return
+
+        if dist_sync_fn is None:
+            # User provided a bool, so we assume DDP if available
+            dist_sync_fn = gather_all_tensors
+
+        synced = False
+        cache = []
+        if self._to_sync and dist_sync_fn is not None:
```

Comment:
```suggestion
        if is_distributed and self._to_sync and dist_sync_fn is not None:
```

Reply:
```suggestion
```
removed in favor of the changed condition below

Reply:
Done !